Update contrib docs to reflect simplified process (#9465)
paritytech/substrate: docs/CONTRIBUTING.adoc

To create a Polkadot companion PR:

. Pull latest Polkadot master (or clone it, if you haven't yet).

> . Override substrate deps to point to your local path or branch using https://github.com/bkchr/diener.

Author: ```suggestion
. Override substrate deps to point to your local path or branch using https://github.com/bkchr/diener. (E.g. `diener patch --crates-to-patch ../substrate --substrate` assuming substrate is cloned to a sister dir to polkadot)
```

> . Override substrate deps to point to your local path or branch using https://github.com/bkchr/diener. (E.g. `diener patch --crates-to-patch ../substrate --substrate` assuming substrate is cloned to a sister dir to polkadot)

Author: ```suggestion
. Override substrate deps to point to your local path or branch using https://github.com/bkchr/diener. (E.g. from the polkadot clone dir run `diener patch --crates-to-patch ../substrate --substrate` assuming substrate clone is in a sibling dir)
```

. Make the changes required and build polkadot locally.

> . Submit all this as a PR against the Polkadot Repo.

Reviewer: this can be confusing, the changes made by diener should not be committed

. Pull latest Polkadot master (or clone it, if you haven't yet).
. Override your local cargo config to point to your local substrate (pointing to your WIP branch): place `paths = ["path/to/substrate"]` in `~/.cargo/config`.
. Make the changes required and build polkadot locally.
. Submit all this as a PR against the Polkadot Repo.
. Link to your Polkadot PR in the _description_ of your _Substrate_ PR as "polkadot companion: [URL]"
. Now you should see that the `check_polkadot` CI job will build your Substrate PR agains the mentioned Polkadot branch in your PR description.
. Someone will need to approve the Polkadot PR before the Substrate CI will go green. (The Polkadot CI failing can be ignored as long as the polkadot job in the _substrate_ PR is green).
. Wait for reviews on both the Substrate and the Polkadot PRs.
. Once the Substrate PR runs green, a member of the `parity` github group can comment on the Substrate PR with `bot merge` which will:
    - Merge the Substrate PR.
    - The bot will push a commit to the Polkadot PR updating its Substrate reference.
    - If the polkadot PR origins from a fork then a project member may need to press `approve run` on the polkadot PR.

>     - The bot will merge the Polkadot PR once all its CI checks are green.

Author: ```suggestion
    - The bot will merge the Polkadot PR once all its CI checks are green.
    
    Note: The merge-bot currently doesn't work with forks on org accounts, only individual accounts. 
```

. Pull latest Polkadot master (or clone it, if you haven't yet).
. Override your local cargo config to point to your local substrate (pointing to your WIP branch): place `paths = ["path/to/substrate"]` in `~/.cargo/config`.
. Make the changes required and build polkadot locally.
. Submit all this as a PR against the Polkadot Repo.
. Link to your Polkadot PR in the _description_ of your _Substrate_ PR as "polkadot companion: [URL]"

> . Link to your Substrate PR in the _description_ of your _Polkadot_ PR as "Companion PR for [URL]"

Reviewer 2: that's not necessary, let's avoid cargo cult here
```suggestion
```

Author: I noticed that if you didn't do that, then the polkadot CI substrate still pointed to master (but we're told to ignore the polkadot CI so does presumably this is true but doesn't matter?). I think the main point we need to get across is to IGNORE the polkadot CI as it will get run in the substrate PR but maybe things are more nuanced then that?

. Now you should see that the `check_polkadot` CI job will build your Substrate PR agains the mentioned Polkadot branch in your PR description.
. Someone will need to approve the Polkadot PR before the Substrate CI will go green. (The Polkadot CI failing can be ignored as long as the polkadot job in the _substrate_ PR is green).
. Wait for reviews on both the Substrate and the Polkadot PRs.
. Once the Substrate PR runs green, a member of the `parity` github group can comment on the Substrate PR with `bot merge` which will:
    - Merge the Substrate PR.
    - The bot will push a commit to the Polkadot PR updating its Substrate reference.

>     - If the polkadot PR origins from a fork then a project member may need to press `approve run` on the polkadot PR.

Reviewer 2: I think there is another caveat, that bot doesn't work with forks on org accounts, instead of individual ones, e.g. https://github.com/paritytech/polkadot/pull/3477 cc @joao-paulo-parity

Reviewer 3: That's right, see https://github.com/isaacs/github/issues/1681#issuecomment-825453532 for context

Reviewer: I don't know where is the button "approve run" but I guess it is correct.

. Pull latest Polkadot master (or clone it, if you haven't yet).

> . Override your local cargo config to point to your local substrate (pointing to your WIP branch): place `paths = ["path/to/substrate"]` in `~/.cargo/config`.

Reviewer: I feel like mentionning `diener` tool would be helpful

. Make the changes required and build polkadot locally.
. Submit all this as a PR against the Polkadot Repo.
. Link to your Polkadot PR in the _description_ of your _Substrate_ PR as "polkadot companion: [URL]"
. Now you should see that the `check_polkadot` CI job will build your Substrate PR agains the mentioned Polkadot branch in your PR description.
. Someone will need to approve the Polkadot PR before the Substrate CI will go green. (The Polkadot CI failing can be ignored as long as the polkadot job in the _substrate_ PR is green).
. Wait for reviews on both the Substrate and the Polkadot PRs.
. Once the Substrate PR runs green, a member of the `parity` github group can comment on the Substrate PR with `bot merge` which will:
    - Merge the Substrate PR.
    - The bot will push a commit to the Polkadot PR updating its Substrate reference.
    - If the polkadot PR origins from a fork then a project member may need to press `approve run` on the polkadot PR.

>     - The bot will merge the Polkadot PR once all its CI checks are green.

Author: Unleash should be temporary - I don't want to mention it. Once frame-metadata gets sorted unleash should work fine.
